piptools/writer.py

```diff
@@ -27,6 +29,8 @@ def write_header(self):
         if self.emit_header:
             yield comment('#')
             yield comment('# This file is autogenerated by pip-compile')
+            if self.write_date:
+                yield comment('# On {} UTC'.format(datetime.utcnow().strftime('%Y-%m-%d %H:%M:%S')))
```

Comment:
I agree UTC is unambiguous, but I'm not sure how _useful_ it is. I think I'd prefer the local date for the user that last generated it. This is for human consumption after all.